shared/AppInsightsCore/src/JavaScriptSDK/InstrumentHooks.ts

```diff
@@ -250,3 +257,23 @@ export function InstrumentFuncs(target:any, funcNames:string[], callbacks: IInst
 
     return hooks;
 }
+
+/**
+ * Add an instrumentation hook to the provided named "event" for the target class / object, this doesn't check whether the
+ * named "event" is in fact a function and just assigns the instrumentation hook to the target[evtName]
+ * @param target - The target object
+ * @param evtName - The name of the event
+ * @param callbacks - The callbacks to configure and call whenever the function is called
+ * @param checkPrototype - If the function doesn't exist on the target should it attempt to hook the prototype function
+ */
+export function InstrumentEvent(target: any, evtName: string, callbacks: IInstrumentHooksCallbacks, checkPrototype:boolean = true): IInstrumentHook {
```

Comment:
Added this "new" function which doesn't "check" if the target is a function and instead just applies the hook to the named "property" of the "target" (which is called from the SDK for events).